### PR TITLE
Remove redundant `eval()` & `train()` in `SleepStagerChambon2018`

### PR DESCRIPTION
**Pull request by OverLordGoldDragon**
They don't affect shapes.

This doesn't seem to belong anywhere in `whats_new`, nor is it important, so I omitted, but feel free to edit.

### CODE REVIEW

File: braindecode/models/sleep_stager_chambon_2018.py [hunk 1]
```diff
@@ -133,10 +133,8 @@ def __init__(
             )
 
     def _len_last_layer(self, n_channels, input_size):
-        self.feature_extractor.eval()
```
Comment (agramfort) [inline braindecode/models/sleep_stager_chambon_2018.py:136]: why do you say it's redundant? layers like drop out need it and it's not the same as no_grad

Comment (OverLordGoldDragon) [reply braindecode/models/sleep_stager_chambon_2018.py:136]: The method operates on output shapes, which are unaffected.

**Comment by agramfort** [comment]
I am not convinced we should merge this one.

**Comment by OverLordGoldDragon** [comment]
> but is it a bad practice to move the model to eval mode when you are not
training?

No, it's just confusing as it implies intent, i.e. that it'd be different if we removed the code. (I found it confusing and had to look into it).

Feel free to proceed as seen fit.

**Pull request closed without merging**